packages/node/src/indexer/indexer.manager.ts

```diff
@@ -44,6 +50,30 @@ export class IndexerManager {
     private eventEmitter: EventEmitter2,
   ) {}
 
+  filterDataSources(): SubqlRuntimeDatasource[] {
+    const dataSourcesFilteredSpecName = this.project.dataSources.filter(
+      (ds) =>
+        !ds.filter?.specName ||
+        ds.filter.specName === this.api.runtimeVersion.specName.toString(),
+    );
+    if (dataSourcesFilteredSpecName.length === 0) {
+      logger.error(
+        `Did not find any dataSource match with network specName ${this.api.runtimeVersion.specName}`,
+      );
+      process.exit(1);
+    }
+    const dataSourcesFilteredStartBlock = dataSourcesFilteredSpecName.filter(
```

Comment:
we still need both checks be put together, for cases that if you check seperatedly they are all pass but together false.